### PR TITLE
Add Documentation for DocumentReference Read and Search by ID

### PR DESCRIPTION
**Pull request by averyallen**
Updates documentation for new DocumentReference read/search by ID functionality.  Also updates example search parameters to be usable values with the open sandbox instance.

We'll need to wait a few days before this can be merged, but I wanted to go ahead and get it out here for review.

### CODE REVIEW

**Comment by daliboz** [comment]
We should add implementation notes for search and read by id to reference the Binary documentation on how to set the Accept header when following the Attachment link. Something like: Contents of the document are found by following the Attachment link. See more information on the Binary resource to determine what Authorization scopes are required, and how to set the Accept header when downloading document contents.

File: content/millennium/dstu2/infrastructure/document-reference.md [hunk 1]
```diff
@@ -51,23 +51,26 @@ Search for DocumentReferences that meet supplied query parameters:
...
+`patient`       | This, or one of `_id` or `subject`     | [`reference`] | The patient to which the document reference belongs. Example: `patient=1316024`
+`subject`       | This, or one of `_id` or `patient`     | [`reference`] | The subject of the document reference. Must represent a Patient resource. May use the :Patient modifier. Example: `subject=Patient/1316024 or subject:Patient=1316024`
+`encounter`     | No                                     | [`reference`] | The encounter to which the document reference belongs.  Must represent an Encounter resource.  May include a single or comma separated list of references. Example: `encounter=1621910`
+`created`       | No                                     | [`date`]      | A date/time the referenced document was created.  Must use the `ge` and `le` prefixes.  Example: `created=le2017-01-5&created=ge2017-02-7`
+[`_count`]      | No                                     | [`number`]    | The maximum number of results to return.
 
 Notes:
 
+- The `_id` parameter may not be provided at the same time as the `patient`, `subject`, `encounter`, `created` or `_count` parameters.
```
Comment (beichhor) [inline content/millennium/dstu2/infrastructure/document-reference.md]: Do you need an Oxford comma here?

File: content/millennium/dstu2/infrastructure/document-reference.md [hunk 2]
```diff
@@ -51,23 +51,26 @@ Search for DocumentReferences that meet supplied query parameters:
 _Implementation Notes_
 
 * Search results are currently limited to published clinical documents.
+* Contents of the document are found by following the Attachment url. See more information on the [Binary resource] to determine what Authorization scopes are required, and how to set the Accept header when downloading document contents.
```
Comment (beichhor) [inline content/millennium/dstu2/infrastructure/document-reference.md]: Should it be URL all caps? Noticing it's url here and URL in your other PR.

File: content/millennium/dstu2/infrastructure/document-reference.md [hunk 3]
```diff
@@ -164,6 +167,38 @@ The common [errors] may be returned. In addition, [OperationOutcomes] may be ret
...
 
+## Retrieve by id
+
+List an individual DocumentReference by its id:
+
+    GET /DocumentReference/:id
+
+_Implementation Notes_
+
+* Contents of the document are found by following the Attachment url. See more information on the [Binary resource] to determine what Authorization scopes are required, and how to set the Accept header when downloading document contents.
```
Comment (beichhor) [inline content/millennium/dstu2/infrastructure/document-reference.md]: Same as above.

**Comment by beichhor** [comment]
+1

**Pull request merged**